hibernate-core/src/main/java/org/hibernate/type/AbstractStandardBasicType.java

```diff
@@ -71,11 +71,7 @@ public T fromStringValue(String xml) throws HibernateException {
 	}
 
 	protected T getReplacement(T original, T target, SharedSessionContractImplementor session) {
-		if ( original == LazyPropertyInitializer.UNFETCHED_PROPERTY ) {
-			return target;
-		}
-		else if ( !isMutable() ||
-					( target != LazyPropertyInitializer.UNFETCHED_PROPERTY && isEqual( original, target ) ) ) {
+		if ( !isMutable() || ( target != null && isEqual( original, target ) ) ) {
```

Comment:
I know you didn't change this, but If the property is mutable, and `isEqual( original, target ) == true`, I think `target` should be returned.